Remotion/Core/Reflection/MethodInfoAdapter.cs

```diff
@@ -57,15 +56,31 @@ private MethodInfoAdapter (MethodInfo methodInfo)
       _methodInfo = methodInfo;
 
       _cachedDeclaringType = new Lazy<ITypeInformation> (
-          () => Maybe.ForValue (_methodInfo.DeclaringType).Select (TypeAdapter.Create).ValueOrDefault(),
+          () =>
+          {
+            var declaringType = _methodInfo.DeclaringType;
+
+            if (declaringType == null)
+              return null;
+
+            return TypeAdapter.Create (declaringType);
+          },

```

Comment:
Are two occurrences of this code part enough to extract it?

Reply:
I don't think so either.

Reply:
Ah yes. Right. There's ONE type of statement that benefits from Maybe even in this day and age. Well, now it's gone :D 

Should we ever feel there's too many calls to TypeAdapter.Create, MethodInfoAdapter.Create, and PropertyInfoAdapter.Create with a null-guard in front, we can always introduce a CreateOrNull Method (name TBD).

Reply:
Okay, there's a bunch of those after all (see below). We will want to seriously consider introducing a nullable alternative to Create. TBD if we should do this before or after these changes here. I would prefer to do them before to reduce noise but it depends on total time this will take.

Reply:
@MichaelKetting I would add them before. Should they be added `public` or `internal`? I thought about names but `CreateOrNull` seems best.

Reply:
public. need tests, too :)

Also, internal's only for APIs that are really needed for cross-class communication of non-public information. uses very rarely. Or when using shared source files to prevent dependency conflics

Name: CreateOrNull it is. Also checked the occurances in rest of code and the alternative of CreateOrDefault isn't really better and we do have a ton of ...OrNull methods in the code anway.

Reply:
@MichaelKetting is the addition of `CreateOrNull` a separate issue?